lib/compat/wordpress-6.2/default-filters.php

```diff
@@ -17,16 +17,24 @@
  * @package gutenberg
  */
 
-add_action( 'switch_theme', 'wp_theme_has_theme_json_clean_cache' );
-add_action( 'start_previewing_theme', 'wp_theme_has_theme_json_clean_cache' );
-add_action( 'upgrader_process_complete', '_wp_theme_has_theme_json_clean_cache_upon_upgrading_active_theme', 10, 2 );
-add_action( 'save_post_wp_global_styles', array( 'WP_Theme_JSON_Resolver_Gutenberg', 'clean_cached_data' ) );
+add_action( 'activated_plugin', 'gutenberg_get_global_settings_clean_cache' );
```

Comment:
This has been sorted so it is easier to scan what is missing. This PR only adds the callbacks `gutenberg_get_global_settings_clean_cache` and `_gutenberg_get_global_settings_clean_cache_upon_upgrading`.

Reply:
Let create one function with all these calls in them, let's not hook theme functions in here, we only need one wrapper.

Reply:
What is that advantage of creating a wrapper?

I see a few disadvantage s: more PHP namespace pollution, lack of ability to unhook functions individually, and I'd even argue maintanibility (I discovered we were missing a couple of hooks due to how this was organized).

Reply:
So instead of one function to clear this cache we have multiple? I suggest having one function and merging `gutenberg_get_global_settings_clean_cache` and `gutenberg_get_global_stylesheet_clean_cache`. 

Having more callback, adds overhead and has a performance drawback. As this functions are unlikely to be unhooked, there is little benefit and mostly downside here.

Reply:
According to the current behavior, it looks like a wrapper makes sense since we're essentially **hooking into the same actions for invalidating 3 different caches**. I think that makes easier to miss adding the needed hooks. However, if we wrap the invalidation of these 3 different caches like the code below, it'd be harder to miss something.

```php
function gutenberg_clean_global_styles_cache() {
  gutenberg_get_global_settings_clean_cache();
  gutenberg_get_global_stylesheet_clean_cache();
  WP_Theme_JSON_Resolver_Gutenberg::clean_cached_data();
}

function _gutenberg_clean_global_styles_cache_upon_upgrading( $upgrader, $options ) {
  _gutenberg_get_global_settings_clean_cache_upon_upgrading( $upgrader, $options );
  _gutenberg_get_global_stylesheet_clean_cache_upon_upgrading( $upgrader, $options );
  WP_Theme_JSON_Resolver_Gutenberg::_clean_cached_data_upon_upgrading( $upgrader, $options );
}

add_action( 'activated_plugin', 'gutenberg_clean_global_styles_cache' );
add_action( 'deactivated_plugin', 'gutenberg_clean_global_styles_cache' );
add_action( 'save_post_wp_global_styles', 'gutenberg_clean_global_styles_cache' );
add_action( 'start_previewing_theme', 'gutenberg_clean_global_styles_cache' );
add_action( 'switch_theme', 'gutenberg_clean_global_styles_cache' );
add_action( 'upgrader_process_complete', '_gutenberg_clean_global_styles_cache_upon_upgrading', 10, 2 );

add_action( 'start_previewing_theme', 'wp_theme_has_theme_json_clean_cache' );
add_action( 'switch_theme', 'wp_theme_has_theme_json_clean_cache' );
add_action( 'upgrader_process_complete', '_wp_theme_has_theme_json_clean_cache_upon_upgrading_active_theme', 10, 2 );
```

Anyway, I wouldn't consider this a blocker for this PR. It can be addressed on a follow-up PR.